commit 08fd9112fa9a9f938b637cbc4f50aa3f12a7013e
Author: Dima Pasechnik <dimpase@gmail.com>
Date:   Sat Feb 3 04:08:10 2018 +0000

    don't do floor(int)
    
    This is another catch from an attempt to build Singular on Solaris 11,
    this time using gcc 5.4.0. It errors out saying
    
    walk.cc:4570:103: error: call of overloaded 'floor(int)' is ambiguous
               (*next_weight2)[i] = 1 + (*curr_weight)[i] + floor(weight_rad*(*next_weight2)[i]/weight_norm);
    
    walk.cc:4574:99: error: call of overloaded 'floor(int)' is ambiguous
               (*next_weight2)[i] = (*curr_weight)[i] + floor(weight_rad*(*next_weight2)[i]/weight_norm);

diff --git a/Singular/walk.cc b/Singular/walk.cc
index 08b7658..0ece307 100644
--- a/Singular/walk.cc
+++ b/Singular/walk.cc
@@ -4566,11 +4566,11 @@ static intvec* MWalkRandomNextWeight(ideal G, intvec* orig_M, intvec* target_wei
       {
         if((*next_weight2)[i] < 0)
         {
-          (*next_weight2)[i] = 1 + (*curr_weight)[i] + floor(weight_rad*(*next_weight2)[i]/weight_norm);
+          (*next_weight2)[i] = 1 + (*curr_weight)[i] + weight_rad*(*next_weight2)[i]/weight_norm;
         }
         else
         {
-          (*next_weight2)[i] = (*curr_weight)[i] + floor(weight_rad*(*next_weight2)[i]/weight_norm);
+          (*next_weight2)[i] = (*curr_weight)[i] + weight_rad*(*next_weight2)[i]/weight_norm;
         }
       }
       if(test_w_in_ConeCC(G,next_weight2) == 1)
